### PR TITLE
Stats: Adding more banner events

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Related to #79339

## Proposed Changes

* adding tracking events for banner dismissal
* adding tracking events for viewing a success banner after purchasing stats

## Testing Instructions

* go to the live branch and trigger the events
* go to the tracking tool and check if an event has been triggered

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-purchase/index.tsx [hunk 1]
```diff
@@ -54,6 +55,10 @@ const StatsPurchasePage = ( { query }: { query: { redirect_uri: string; from: st
 		return isProductOwned( siteProducts, PRODUCT_JETPACK_STATS_PWYW_YEARLY );
 	}, [ siteProducts ] );
 
+	if ( query?.from && query.from === 'jetpack-my-jetpack' ) {
```
Comment (kangzj) [inline client/my-sites/stats/stats-purchase/index.tsx]: We've got `from` recorded with the page view of the page, so I guess we could omit this one?

https://github.com/Automattic/wp-calypso/blob/f3aedf8a17acc280bdfc9036d6b0a4b65434bbc9/client/my-sites/stats/stats-purchase/index.tsx#L86

File: client/my-sites/stats/stats-notices/free-plan-purchase-success-notice.tsx [hunk 2]
```diff
@@ -23,6 +24,14 @@ const FreePlanPurchaseSuccessJetpackStatsNotice = ( { siteId }: StatsNoticeProps
 		setNoticeDismissed( true );
 	};
 
+	useEffect( () => {
+		if ( ! noticeDismissed ) {
+			isOdysseyStats
+				? recordTracksEvent( 'jetpack_odyssey_stats_purchase_success_notice_viewed' )
+				: recordTracksEvent( 'calypso_stats_purchase_success_notice_viewed' );
+		}
+	}, [ noticeDismissed, isOdysseyStats ] );
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-notices/free-plan-purchase-success-notice.tsx]: We could probably omit this. Context: pejTkB-K8-p2#comment-677

File: client/my-sites/stats/stats-purchase/stats-purchase-wizard.jsx [hunk 3]
```diff
@@ -81,6 +86,8 @@ const ProductCard = ( { siteSlug, siteId, commercialProduct, pwywProduct, redire
 	// change the plan to commercial on the personal plan confirmation
 	const handlePlanSwap = ( e ) => {
 		e.preventDefault();
+		recordTracksEvent( `calypso_stats_plan_switched_from_personal_to_commercial` );
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-purchase/stats-purchase-wizard.jsx:89]: (Not blocking) @Zubie04 asked a good question: Should there be a commercial-to-personal equivalent to this event?

My follow-up question if yes: Could we generalize this event to `calypso_stats_plan_type_switched` in that case?

File: client/my-sites/stats/stats-notices/paid-plan-purchase-success-notice.tsx [hunk 4]
```diff
@@ -13,6 +14,14 @@ const PaidPlanPurchaseSuccessJetpackStatsNotice = () => {
 		setNoticeDismissed( true );
 	};
 
+	useEffect( () => {
+		if ( ! noticeDismissed ) {
+			isOdysseyStats
+				? recordTracksEvent( 'jetpack_odyssey_stats_purchase_success_notice_viewed' )
+				: recordTracksEvent( 'calypso_stats_purchase_success_notice_viewed' );
+		}
+	}, [ noticeDismissed, isOdysseyStats ] );
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-notices/paid-plan-purchase-success-notice.tsx]: We could probably omit this. Context: pejTkB-K8-p2#comment-677

Comment (jsnmoon) [reply client/my-sites/stats/stats-notices/paid-plan-purchase-success-notice.tsx]: Omitted via [c9c6c09](https://github.com/Automattic/wp-calypso/pull/79894/commits/c9c6c09c2576ed47c28a94c42973aea0b5f7bad8)

File: client/my-sites/stats/stats-notices/free-plan-purchase-success-notice.tsx (hunk 2 above)
Comment (jsnmoon) [reply client/my-sites/stats/stats-notices/free-plan-purchase-success-notice.tsx]: Omitted via [c9c6c09](https://github.com/Automattic/wp-calypso/pull/79894/commits/c9c6c09c2576ed47c28a94c42973aea0b5f7bad8)

**Review by jsnmoon — APPROVED** [review]
This looks good! We can iterate on feedback in a future PR as necessary.

**Pull request merged**